### PR TITLE
feat: added component default values

### PR DESCRIPTION
**Pull request by Commandserver**
## Code change checklist

- [x] I have ensured that all methods and functions are **fully documented** using doxygen style comments.
- [x] My code follows the [coding style guide](https://dpp.dev/coding-standards.html).
- [x] I tested that my change works before raising the PR.
- [x] I have ensured that I did not break any existing API calls.
- [x] I have not built my pull request using AI, a static analysis tool or similar without any human oversight.

### CODE REVIEW

File: src/dpp/user.cpp [hunk 1]
```diff
@@ -126,6 +126,16 @@ std::string user::get_default_avatar_url() const {
 	}
 }
 
+std::string user::get_avatar_decoration_url(uint16_t size) const {
+	if (this->id) {
+		return utility::cdn_endpoint_url_hash({ i_png },
+											  "avatar-decorations/" + std::to_string(this->id), this->avatar_decoration.to_string(),
```
Comment (Jaskowicz1) [inline src/dpp/user.cpp:132]: indentation?

**Comment by Jaskowicz1** [comment]
Wait, 4 commits from here are in #732 .
How has GitHub not picked up on this?

**Comment by braindigitalis** [comment]
> Wait, 4 commits from here are in #732 . How has GitHub not picked up on this?

It's just a difference in history between branches, it'll iron out in the merge

**Comment by Jaskowicz1** [comment]
> > Wait, 4 commits from here are in #732 . How has GitHub not picked up on this?
> 
> It's just a difference in history between branches, it'll iron out in the merge

Ah right 👍

**Comment by Commandserver** [comment]
Oopsy whoopsy

**Review by Mishura4 — APPROVED** [review]

**Comment by braindigitalis** [comment]
is this ready at all?

**Comment by Jaskowicz1** [comment]
I've renamed this PR from `feat: added message_author_id field to message_reaction_add_t event` to `feat: added component default values`. I'll also test this and see if this good to go 👍

File: src/dpp/message.cpp [hunk 2]
```diff
@@ -345,6 +353,15 @@ void to_json(json& j, const component& cp) {
 		if (cp.max_values >= 0) {
 			j["max_values"] = cp.max_values;
 		}
+		if (!cp.default_values.empty()) {
+			j["default_values"] = json::array();
+			for (auto const &v : cp.default_values) {
+				json o;
+				o["id"] = v.id;
+				o["type"] = v.type;
```
Comment (Jaskowicz1) [inline src/dpp/message.cpp:361]: This isn't right, discord doesn't take an int for the type but rather a string. Change `o["type"] = v.type;` to be something like this:

```c++
if(v.type == dpp::cdt_role) {
	o["type"] = "role";
} else if(v.type == dpp::cdt_channel) {
	o["type"] = "channel";
} else if(v.type == dpp::cdt_user) {
	o["type"] = "user";
}
```

File: src/dpp/message.cpp [hunk 3]
```diff
@@ -363,6 +380,15 @@ void to_json(json& j, const component& cp) {
 				j["channel_types"].push_back(type);
 			}
 		}
+		if (!cp.default_values.empty()) {
+			j["default_values"] = json::array();
+			for (auto const &v : cp.default_values) {
+				json o;
+				o["id"] = v.id;
+				o["type"] = v.type;
```
Comment (Jaskowicz1) [inline src/dpp/message.cpp:388]: ditto comment on line 361

**Review by Jaskowicz1 — CHANGES_REQUESTED** [review]

**Comment by braindigitalis** [comment]
i have ticked the boxes in the pr, and added archies changes to merge this, as i need it in my own projects.

**Pull request merged**